src/main/java/randoop/generation/ConstantMiningSelection.java

```diff
@@ -0,0 +1,103 @@
+package randoop.generation;
+
+import java.util.HashMap;
+import java.util.Map;
+import randoop.main.GenInputsAbstract;
+import randoop.sequence.Sequence;
+import randoop.util.Randomness;
+import randoop.util.SimpleList;
+
+/**
+ * Implements the Constant Mining component, as described by the paper "GRT: Program-Analysis-Guided
+ * Random Testing" by Ma et. al (appears in ASE 2015):
+ * https://people.kth.se/~artho/papers/lei-ase2015.pdf .
+ */
+public class ConstantMiningSelection implements InputSequenceSelector {
+  /**
+   * Map of extracted literal sequences to their static weights. These weights are never changed
+   * once initialized.
+   */
+  private final Map<Sequence, Double> literalWeightMap = new HashMap<>();
+
+  /**
+   * Initialize Constant Mining selection by computing weights for literals that appear in classes
+   * under test.
+   *
+   * @param componentManager component generator from {@link ForwardGenerator} used for getting the
+   *     frequency of a literal
+   * @param numClasses number of classes under test
+   * @param literalTermFrequencies a map from a literal to the number of times it appears in any
+   *     class under test. If literalTermFrequencies is null, we simply don't compute any weights to
+   *     put in literalWeightMap and every sequence will end up using its default weight computed
```

Comment:
If this constructor is invoked we expect that literalTermFrequencies is never null. We check for this before the only invocation of ConstantMiningSelection's constructor call in ForwardGenerator. Thus, I've removed the comment regarding default weights but didn't comment on a null case since we don't expect it to happen.